### ISSUE TITLE
Measuring similarity for a reconstructed molecule

### ISSUE THREAD

**Issue by sgalkina** [THUNLP-MT/PS-VAE#8]
Dear authors, 

Great work! Thanks for making the code documented and available.

I am trying to evaluate the similarity of molecule reconstructions in a basic VAE fashion: inputting a molecule, getting a latent space vector, reconstructing it, measure the similarity with a Morgan fingerprint distance. Something along those lines:

```
z = model.get_z_from_mol(ori_mol)
mol = model.inference_single_z(z, args.max_atom_num, args.add_edge_th, args.temperature)
similarity(ori_mol, mol)
```

However, the reconstruction quality measured in that way, using the pretrained model for ZINC dataset, is not very high (~0.3 on average). The input and output molecules mostly look like this (first column is an input from ZINC test set, the rest of the columns are reconstructions from sampled z vectors for this molecule):
 
![Screenshot 2023-05-24 at 10 59 57](https://github.com/THUNLP-MT/PS-VAE/assets/4077686/3205cbd1-30a8-424b-a16b-73867923b76d)

Another way to reconstruct a similar molecule would be to train a Guacamol predictor with `similarity` or `rediscovery` benchmark function. However, to do this for a whole test set is a very computationally heavy task, that has to be redone every time for a new test set.

Anything I'm missing? How does one reconstruct a molecule? 

Best, 
Svetlana

**Comment by kxz18** [comment]
Hi, thanks for your interest in our work. I think your codes are right. I haven't test the reconstruction ability because the benchmarks are mainly goal-directed generative tasks. I think there is a trade-off between the reconstruction ability and the goal-directed generative ability. As you can see, the training command sets alpha (the weight of reconstruction loss) to 0.1 and beta (the weight of KL loss) to 0.01. This facilitates predicting properties from the latent space, which is good for gradient optimization on the goal-directed tasks, but will harm the reconstruction ability. Therefore, maybe the provided checkpoint is just not so powerful at reconstruction. I think training a new model on the ZINC with larger alpha and tuned beta (the choice of beta is tricky in VAE) might improve the reconstruction ability.

**Comment by sgalkina** [comment]
Thank you for the swift reply! Will try re-training the model
